src/common/misc.c

```diff
@@ -684,18 +685,15 @@ daos_dti_gen(struct dtx_id *dti, bool zero)
 	if (zero) {
 		memset(dti, 0, sizeof(*dti));
 	} else {
-		if (uuid_is_null(dti_uuid))
-			uuid_generate(dti_uuid);
-
```

Comment:
i think the problem is here. i have added some debug PR:
https://github.com/daos-stack/daos/pull/16573
and it shows that there is a chance the we get NULL uuid here which sounds wrong:
https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos/job/PR-16573/1/artifact/Functional%20Hardware%20Medium%20MD%20on%20SSD/ior/small.py/stacktraces.hdr-148/'core.dfuse%20worker.1751965967.102344.stacktrace'/*view*/

Reply:
since dti_uuid is thread local, we end up in this situation. i will push a quick fix to this PR since FY is out

Reply:
It is expected that daos_eq_lib_init() will be called when each thread is created, so such expectation or assumption may be false? @mchaarawi

Reply:
> It is expected that daos_eq_lib_init() will be called when each thread is created, so such expectation or assumption may be false? @mchaarawi

no it is not expected for new thread, so it's false. it is expected for new process (fork), which is why we register at_fork callback in daos.

Reply:
Then it maybe not enough, different threads may generate the same transaction ID if without calling daos_dti_init() in advance.

Reply:
right, so does the change i made to the PR work? 
it passes the test..